src/main/java/openmodularturrets/tileentity/turretbase/TurretBase.java

```diff
@@ -555,29 +547,27 @@ public void updateEntity() {
             this.storage.setCapacity(getMaxEnergyStorageWithExtenders());
 
             // Thaumcraft
-            if (ModCompatibility.ThaumcraftLoaded && TurretHeadUtil.hasPotentiaUpgradeAddon(this)) {
-                if (amountOfPotentia > 0.05F && !(storage.getMaxEnergyStored() - storage.getEnergyStored() == 0)) {
-                    if (VisNetHandler.drainVis(worldObj, xCoord, yCoord, zCoord, Aspect.ORDER, 5) == 5) {
-                        this.amountOfPotentia = this.amountOfPotentia - 0.05F;
-                        this.storage.modifyEnergyStored(Math.round(ConfigHandler.getPotentiaToRFRatio() * 5));
-                    } else {
-                        this.amountOfPotentia = this.amountOfPotentia - 0.05F;
-                        this.storage.modifyEnergyStored(Math.round(ConfigHandler.getPotentiaToRFRatio() / 2));
-                    }
+            if (ModCompatibility.ThaumcraftLoaded && TurretHeadUtil.hasPotentiaUpgradeAddon(this)
+                    && (storage.getMaxEnergyStored() - storage.getEnergyStored()
+                            <= ConfigHandler.getPotentiaToRFRatio() * 5)) {
+                if (VisNetHandler.drainVis(worldObj, xCoord, yCoord, zCoord, Aspect.ORDER, 5) == 5) {
+                    this.storage.modifyEnergyStored(ConfigHandler.getPotentiaToRFRatio() * 5);
+                } else if (this.amountOfPotentia > 0.05F) {
+                    this.amountOfPotentia = this.amountOfPotentia - 0.05F;
+                    this.storage.modifyEnergyStored(ConfigHandler.getPotentiaToRFRatio() * 5);
                 }
```

Comment:
Why did you break whole logic of extracting essentia? I posted correct code for you in my previous review, so please use it.

Reply:
Your code broke the class entirely. I see a bug in there, lemme fix that.

Reply:
Updated the code. Merged the 2 parts of the thaumcraft logic. It now tries to get potentia essence and then proceeds with trying to get ordo vis and boost energy generated 10 fold, else just use up potentia with the usual rate / 2.